CMakeLists.txt

```diff
@@ -1045,6 +1045,10 @@ if(SDL_LIBC)
     check_symbol_exists(elf_aux_info "sys/auxv.h" HAVE_ELF_AUX_INFO)
     check_symbol_exists(poll "poll.h" HAVE_POLL)
 
+    if (NOT CMAKE_SYSTEM_NAME MATCHES "Linux")
+      check_symbol_exists(arc4random_buf "stdlib.h" HAVE_ARC4RANDOM)
```

Comment:
I'm okay with always detecting the symbol, but would guard it in c using similar logic seen in https://github.com/php/php-src/pull/10390

Reply:
fair point but I thought of the case if it was, one day, outside f just SDL_malloc, then just dismiss it only one place..

Reply:
Perhaps do this in SDL_build_config.h.cmake instead?